internal/pkg/deploy/pipeline.go

```diff
@@ -523,43 +533,74 @@ func (stg *PipelineStage) EnvManagerRoleARN() string {
 
 // Test returns a test for the stage.
 // If the stage does not have any test commands, then returns nil.
-func (stg *PipelineStage) Test() *TestCommandsAction {
+func (stg *PipelineStage) Test() (*TestCommandsAction, error) {
 	if len(stg.testCommands) == 0 {
-		return nil
+		return nil, nil
 	}
 
 	var prevActions []orderedRunner
-	for _, deployment := range stg.Deployments() {
-		prevActions = append(prevActions, &deployment)
+	deployActions, err := stg.Deployments()
+	if err != nil {
+		return nil, err
+	}
+	for _, action := range deployActions {
+		prevActions = append(prevActions, &action)
```

Comment:
this will fill `prevActions` with the same action (the last one in deployActions). gotta do something like
```
for i := range deployActions {
    prevActions = append(prevActions, &deployActions[i])
}
```

Reply:
🤦 tyyy!